### PR TITLE
[WIP] Feature 01: MongoDB Implementation

### PR DESCRIPTION
**Pull request by JeewakaSankalpa**
- [x] closes #1 
- [x] tests added / passed
- [x] passes ``git diff upstream/master | flake8 --diff``

NOTE: Be descriptive in your commit name. I don't care about summary, but, in your commit title, it should be summary of what you did.

### CODE REVIEW

**Comment by jaehoonhwang** [comment]
hmm actually can you create test folder inside your db folder
Something like `source/db/tests`
Also, you need to use `pytest` not `unittest`

**Comment by JeewakaSankalpa** [comment]
ok i'll make the changes.

**Comment by JeewakaSankalpa** [comment]
most of the basic structure is done now. Only need to extend the insert function to accommodate markdown files.

**Comment by jaehoonhwang** [comment]
Okay.

**Comment by jaehoonhwang** [comment]
Please update your put your files into `captslog/db`

**Comment by JeewakaSankalpa** [comment]
I have not been well since last sunday so didn't have much time to work on anything. I'll have the  rest of it fixed by this Sunday. Sorry about the Delay

File: source/db/DBHandler.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+class DBHandlerClass:
+    client = MongoClient()
+    db = client['Captains_Log_DB']
+
+    def __init__(self):
+        client = MongoClient()
+        self.db = client['Captains_Log_DB']
+
+    def insert_to_user_table(self, name, username, user_password):
```
Comment (jaehoonhwang) [inline source/db/DBHandler.py]: Don't use `_` for functions
Please use camel case or something alone the line of `insertUsertable`

File: source/db/DBHandler.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        Args:
+            title (str): Title of the Journel Entry
+            tags (list): Tangs for the Entry
+            content(str): Content of the Entry
+
+        return:
+            int: 1 for success or 0 for failure
+
+        """
+        if title == str(""):
```
Comment (jaehoonhwang) [inline source/db/DBHandler.py]: This should be clean up by using `if title`

File: source/db/DBHandler.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            _id:  Object ID of the entry you want to change
+
+        Return:
+            bool:1 if the delete was successful. 0 if it fails
+        """
+
+        entries_table = self.db["Entries_Table"]
+        if not entries_table.find_one({"_id": ObjectId(_id)}):
+            # print "The specified entry does not Exist"
+            return 0
```
Comment (jaehoonhwang) [inline source/db/DBHandler.py]: It should be `return False` for readability

File: source/db/DBHandler.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        entries_table.delete_one({"_id": ObjectId(_id)})
+        return 1
```
Comment (jaehoonhwang) [inline source/db/DBHandler.py]: Same thing here `return True`

**Comment by JeewakaSankalpa** [comment]
Travis CI check fails because some test cases fails. Those test cases needs to check the mongoDB files and return the correct values. Without having an instance of MongoDB running these tests will always fail.

**Comment by jaehoonhwang** [comment]
Use `mock` package.

**Comment by JeewakaSankalpa** [comment]
Haven't used it before i'll look into it now
Thanks

File: source/db/DBHandler.py (hunk 3 above)
Comment (JeewakaSankalpa) [inline source/db/DBHandler.py]: Fixed

File: source/db/DBHandler.py (hunk 4 above)
Comment (JeewakaSankalpa) [inline source/db/DBHandler.py]: Fixed

File: source/db/DBHandler.py (hunk 1 above)
Comment (JeewakaSankalpa) [inline source/db/DBHandler.py]: As the PEP8 check fails if we use CamelCase, decided to go with _

**Comment by JeewakaSankalpa** [comment]
I implemented the test cases without using the mock package by implementing exception handlers to handle the cases when the Server wasn't found.  It works fine now. 

But the draw back is there is now way to achieve a high code coverage without running the test cases while the server is both running and not running.  I might be able to simulate this in my local environment by creating and running a script to run the server for the first half of the test cases and the rest without the server if you want me to. But not sure how to implement it in here with TravisCI.

I might be able to boost the percentage a bit more if i look into it more but that probably won't boost it up to 85%

**Comment by jaehoonhwang** [comment]
Mock library provides way to "simulate" the server.
If you can get it to at least 80%, that would be nice.

**Comment by jaehoonhwang** [comment]
https://github.com/mongomock/mongomock

**Comment by JeewakaSankalpa** [comment]
That last one is helpful thanks. I think i can get it past 85% with it. But I might not be able to do it now before the the 10 pm deadline. I got it closer to 80% for now. Will keep working on it later probably tomorrow.

**Comment by jaehoonhwang** [comment]
Looks good merging it

**Pull request merged**